### PR TITLE
chore(#1426): Add dev pages for structure and nav components

### PR DESCRIPTION
**Pull request by amar-ABgov**

### CODE REVIEW

File: src/routes/components/FormStepper.tsx [hunk 1]
```diff
@@ -1,62 +1,402 @@
...
+      type: "number",
+      description:
+        "The current step state value. Leaving it blank will allow any step to be accessed.",
+    },
+  ];
+
+  const formStepProperties: ComponentProperty[] = [
+    {
+      name: "text",
+      type: "string",
```
Comment (ArakTaiRoth) [inline src/routes/components/FormStepper.tsx:50]: You need to add `required: true,`

File: src/routes/components/Pagination.tsx [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+      defaultValue: "10",
+      description: "Number of data items shown per page",
+    },
+    {
+      name: "variant",
+      type: "all | links-only",
+      required: false,
+      defaultValue: "all",
+      description: "Controls which nav controls are visible",
+    },
```
Comment (ArakTaiRoth) [inline src/routes/components/Pagination.tsx:91]: You're missing the `onChange` and `_change` properties for React and Angular

File: src/components/component-header/ComponentHeader.tsx [hunk 3]
```diff
@@ -21,7 +23,7 @@ export const ComponentHeader: React.FC<Props> = (props: Props) => {
     <div className="component-header">
       <GoABadge type="information" content={props.category} />
       <h1>{props.name}</h1>
-      <h3>{props.description}</h3>
+      <h3><Markup content={props.description} /></h3>
```
Comment (chrisolsen) [inline src/components/component-header/ComponentHeader.tsx:26]: Instead of adding an external library for this, let's try the following first and if required create a component for it.
https://legacy.reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml

File: src/components/sandbox/Sandbox.tsx [hunk 4]
```diff
@@ -195,12 +195,11 @@ function ComponentList(props: ComponentListProps): ReactElement[] {
...
-    React.isValidElement(el)
-    && typeof el.type === 'function'
-    && (el.type.name.toLowerCase().startsWith(props.type)
-        || props.sandboxProps.allow?.includes(el.type.name)
+    React.isValidElement(el) && (
+      (typeof el.type === "function" && el.type.name.toLowerCase().startsWith(props.type)) ||
+      (typeof el.type === "string" && props.sandboxProps.allow?.includes(el.type))
     )
-  );
+  )
```
Comment (chrisolsen) [inline src/components/sandbox/Sandbox.tsx:202]: What is the reason for the change in logic here?

File: src/App.tsx [hunk 5]
```diff
@@ -47,6 +47,7 @@ import IconsPage from "@routes/components/Icons";
 import ListPage from "@routes/components/List";
 import ModalPage from "@routes/components/Modal";
 import NotificationBannerPage from "@routes/components/Notificationbanner";
+import PaginationPage from './routes/components/Pagination';
```
Comment (chrisolsen) [inline src/App.tsx:50]: Just missed using the `@` here

File: src/App.tsx [hunk 6]
```diff
@@ -114,6 +120,8 @@ const router = createBrowserRouter(
       >
         <Route index element={<AllComponentsPage />} />
         <Route path="accordion" element={<AccordionPage />} />
+        <Route path="header" element={<AppHeaderPage />} />
+        <Route path="footer" element={<AppFooterPage />} />
         <Route path="badge" element={<BadgePage />} />
         <Route path="badge" element={<BadgePage />} />
```
Comment (chrisolsen) [inline src/App.tsx:126]: Not your code, but there are two routes to `BadgePage`

File: src/App.tsx [hunk 7]
```diff
@@ -114,6 +120,8 @@ const router = createBrowserRouter(
       >
         <Route index element={<AllComponentsPage />} />
         <Route path="accordion" element={<AccordionPage />} />
+        <Route path="header" element={<AppHeaderPage />} />
+        <Route path="footer" element={<AppFooterPage />} />
```
Comment (chrisolsen) [inline src/App.tsx:124]: Should these be `app-header` and `app-footer`?

File: src/routes/components/FormStepper.tsx [hunk 8]
```diff
@@ -1,62 +1,415 @@
...
+  GoAPages,
+  GoASkeleton,
+  GoASpacer,
+  GoATab,
+  GoATabs,
+} from "@abgov/react-components";
 import { CodeSnippet } from "@components/code-snippet/CodeSnippet";
+import { ComponentBinding, Sandbox } from "@components/sandbox";
+import { useState } from "react";
+import { Category, ComponentHeader } from "../../components/component-header/ComponentHeader";
```
Comment (chrisolsen) [inline src/routes/components/FormStepper.tsx:18]: Can use `@components/...` here

File: src/routes/components/FormStepper.tsx [hunk 9]
```diff
@@ -1,62 +1,415 @@
...
+import {
+  ComponentProperties,
+  ComponentProperty,
+} from "../../components/component-properties/ComponentProperties";
```
Comment (chrisolsen) [inline src/routes/components/FormStepper.tsx:22]: Same here

File: src/components/sandbox/Sandbox.tsx (hunk 4 above)
Comment (chrisolsen) [reply src/components/sandbox/Sandbox.tsx:202]: It's fine as is. It was just when I was first looking at the code in split view it was hard to read and looked a lot more complex than it was.

**Review by chrisolsen — APPROVED** [review]

**Review by chrisolsen — APPROVED** [review]
